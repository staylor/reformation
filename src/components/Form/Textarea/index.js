import React, { Component } from 'react';
import { cx } from 'pretty-lights';
import { textareaClass } from './styled';

/* eslint-disable react/prop-types */

// type Props = {
//   value: string,
//   className?: string | null,
//   onChange?: (value: string) => void,
//   bindRef?: (element: any) => void,
// };

export default class Textarea extends Component {
  onChange = e => {
    const value = e.target.value || '';
    if (this.props.onChange) {
      this.props.onChange(value);
    }
  };

  render() {
    const { className, onChange, value, bindRef, ...props } = this.props;
    return (
      <textarea
        {...props}
        ref={bindRef}
        className={cx(textareaClass, className)}
        onChange={this.onChange}
        defaultValue={value}
      />
    );
  }
}

Textarea.defaultProps = {
  className: null,
  onChange: () => null,
  bindRef: () => null,
};
