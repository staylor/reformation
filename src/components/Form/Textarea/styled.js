import { css } from 'pretty-lights';
import { inputBaseClass } from 'components/Form/styled';

// eslint-disable-next-line
export const textareaClass = css`
  ${inputBaseClass};
  display: block;
  height: 80px;
  padding: 2px 6px;
  resize: vertical;
  width: 100%;
`;
