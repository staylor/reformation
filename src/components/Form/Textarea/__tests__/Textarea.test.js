import React from 'react';
import Textarea from 'components/Form/Textarea';
import { shallow } from 'enzyme';

const TEXT_VALUE = 'Run for the border.';

describe('Textarea', () => {
  test('empty', () => {
    const wrapper = shallow(<Textarea />);

    expect(wrapper).toMatchSnapshot();
  });

  test('add className', () => {
    const wrapper = shallow(<Textarea className="foo" />);

    expect(wrapper).toMatchSnapshot();
  });

  describe('onChange', () => {
    test('adding text', () => {
      const func = jest.fn();
      const wrapper = shallow(<Textarea onChange={func} />);
      const value = TEXT_VALUE;
      wrapper.simulate('change', { target: { value } });

      expect(func).toHaveBeenCalledWith(value);
    });

    test('removing text', () => {
      const func = jest.fn();
      const value = TEXT_VALUE;
      const wrapper = shallow(<Textarea onChange={func} value={value} />);
      wrapper.simulate('change', { target: { value: '' } });

      expect(func).toHaveBeenCalledWith('');
    });

    test('bad event', () => {
      const func = jest.fn();
      const value = TEXT_VALUE;
      const wrapper = shallow(<Textarea onChange={func} value={value} />);
      wrapper.simulate('change', { target: { foo: 'bar' } });

      expect(func).toHaveBeenCalledWith('');
    });

    test('bad type', () => {
      const func = jest.fn();
      const value = TEXT_VALUE;
      const wrapper = shallow(<Textarea onChange={func} value={value} />);
      wrapper.simulate('change', { target: { value: false } });

      expect(func).toHaveBeenCalledWith('');
    });
  });
});
