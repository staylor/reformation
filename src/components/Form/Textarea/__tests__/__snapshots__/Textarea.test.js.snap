// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea add className 1`] = `
.lights-0 {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 0;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: border-box;
  color: #32373c;
  font-size: 14px;
  outline: none;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  display: block;
  height: 80px;
  padding: 2px 6px;
  resize: vertical;
  width: 100%;
}

<textarea
  className="lights-0 foo"
  onChange={[Function]}
/>
`;

exports[`Textarea empty 1`] = `
.lights-0 {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 0;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: border-box;
  color: #32373c;
  font-size: 14px;
  outline: none;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  display: block;
  height: 80px;
  padding: 2px 6px;
  resize: vertical;
  width: 100%;
}

<textarea
  className="lights-0"
  onChange={[Function]}
/>
`;
